publications/Multi-stage/flops/estimate_flops.py

```diff
@@ -0,0 +1,135 @@
+"""Script to estimate FLOPs of multi-stage clustering."""
+import os
+import csv
+import numpy as np
+from sklearn.cluster import AgglomerativeClustering
+from pypapi import events, papi_high as high
+import spectralcluster
+
+
+INF = float("inf")
+
+
+def run_flops_spectral(N, D, autotune=True):
+    """Estimate FLOPs of spectral clustering algorithm.
+
+    Args:
+        N: number of inputs
+        D: input dimension
+        autotune: whether to use autotune or not,
+            see https://arxiv.org/abs/2003.02405
+
+    Returns:
+        FLOPs
+    """
+    if N == 0:
+        return 0
+
+    X = np.random.rand(N, D)
+    clusterer = spectralcluster.configs.icassp2018_clusterer
```

Comment:
For autotune here, I have been running [0.8, 0.95, 0.01] instead of the one used in the paper.